pkg/daemon/certificate_writer.go

```diff
@@ -0,0 +1,86 @@
+package daemon
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+)
+
+func (dn *Daemon) handleControllerConfigEvent(obj interface{}) {
+	controllerConfig := obj.(*mcfgv1.ControllerConfig)
+	glog.V(4).Infof("Updating ControllerConfig %s", controllerConfig.Name)
+	dn.enqueueControllerConfig(controllerConfig)
+}
+
+func (dn *Daemon) enqueueControllerConfig(controllerConfig *mcfgv1.ControllerConfig) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(controllerConfig)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("couldn't get key for object %#v: %w", controllerConfig, err))
+		return
+	}
+	dn.ccQueue.AddRateLimited(key)
+}
+
+func (dn *Daemon) controllerConfigWorker() {
+	for dn.processNextControllerConfigWorkItem() {
+	}
+}
+
+func (dn *Daemon) processNextControllerConfigWorkItem() bool {
+	key, quit := dn.ccQueue.Get()
+	if quit {
+		return false
+	}
+	defer dn.ccQueue.Done(key)
+
+	err := dn.syncControllerConfigHandler(key.(string))
+	dn.handleControllerConfigErr(err, key)
+
+	return true
+}
+
+func (dn *Daemon) handleControllerConfigErr(err error, key interface{}) {
+	if err == nil {
+		dn.ccQueue.Forget(key)
+		return
+	}
+
+	if err := dn.updateErrorState(err); err != nil {
+		glog.Errorf("Could not update annotation: %v", err)
+	}
+	// This is at V(2) since the updateErrorState() call above ends up logging too
+	glog.V(2).Infof("Error syncing ControllerConfig %v (retries %d): %v", key, dn.ccQueue.NumRequeues(key), err)
+	dn.ccQueue.AddRateLimited(key)
+}
+
+func (dn *Daemon) syncControllerConfigHandler(key string) error {
```

Comment:
Basically this will effectively instantly synchronize changes to these certs across all nodes?  I think that's probably fine for this use case, but it's likely at some point that we will want some tracking and testing of rollouts etc.

Reply:
Maybe at least some annotation on the node that says which controllerconfig it has synchronized to?

Reply:
Will consider adding that :+1: